chromium_src/components/payments/content/payment_request.cc

```diff
@@ -0,0 +1,116 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/payments/content/payment_request.h"
+
+#include "base/bind.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/components/brave_rewards/common/constants.h"
+#include "chrome/browser/profiles/profile.h"
+#include "third_party/blink/public/mojom/payments/payment_request.mojom.h"
+
+using payments::mojom::PaymentErrorReason;
+
+#define Init Init_ChromiumImpl
+#include "../../../../../components/payments/content/payment_request.cc"
+#undef Init
+
+namespace payments {
+
+#if !defined(OS_ANDROID)
```

Comment:
why are there so many OS_ANDROID guards? Is this matching upstream guards or is this something we should be adding a payments buildflag for?